File: backend/models/repository.py
from pydantic import BaseModel
from typing import Optional


class Repository(BaseModel):
    id: int
    name: str
    full_name: str
    description: Optional[str] = None
    custom_description: Optional[str] = None
    image_url: Optional[str] = None
    video_url: Optional[str] = None
    deploy_url: Optional[str] = None
    html_url: str
    language: Optional[str] = None
    stargazers_count: int = 0
    forks_count: int = 0
    topics: list[str] = []
    updated_at: str
    is_featured: bool = False
    is_premium_only: bool = False

File: frontend/src/types/repository.ts
export interface Repository {
  id: number;
  name: string;
  full_name: string;
  description: string | null;
  custom_description?: string;
  image_url?: string;
  video_url?: string;
  deploy_url?: string;
  html_url: string;
  language: string | null;
  stargazers_count: number;
  forks_count: number;
  topics: string[];
  updated_at: string;
  is_featured: boolean;
  is_premium_only?: boolean;
}
